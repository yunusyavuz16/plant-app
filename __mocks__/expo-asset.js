module.exports = {
  Asset: {
    loadAsync: jest.fn(),
  },
};