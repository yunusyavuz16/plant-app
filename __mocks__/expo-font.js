module.exports = {
  useFonts: () => [true, null],
};