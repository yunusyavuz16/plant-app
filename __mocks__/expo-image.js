const { View } = require("react-native");

module.exports = {
  Image: (props) => <View testID="mock-image" {...props} />,
};