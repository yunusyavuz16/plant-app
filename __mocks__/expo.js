module.exports = {
  __esModule: true,
  default: {
    registerRootComponent: jest.fn(),
  },
};