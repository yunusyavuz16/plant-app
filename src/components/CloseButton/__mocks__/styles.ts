export const mockStyles = {
  closeButton: {
    width: 32,
    height: 32,
    borderRadius: 16,
    backgroundColor: '#000000',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    top: 17,
    right: 17,
    zIndex: 10,
  },
  closeText: {
    color: '#FFFFFF',
    fontSize: 24,
    textAlign: 'center',
    fontWeight: '400',
  },
};

export const mockTheme = {
  colors: {
    overlay: {
      dark: '#000000',
    },
    text: {
      white: '#FFFFFF',
    },
  },
  layout: {
    icon: {
      sm: 32,
    },
  },
  spacing: {
    xl: 16,
    xxl: 24,
  },
  typography: {
    size: {
      xl: 24,
    },
  },
};