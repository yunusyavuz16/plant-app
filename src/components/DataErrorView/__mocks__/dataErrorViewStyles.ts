export const mockTheme = {
  colors: {
    background: {
      white: '#FFFFFF',
      lightGray: '#F5F5F5',
    },
    text: {
      primary: '#000000',
      secondary: '#666666',
    },
  },
  spacing: {
    md: 12,
    lg: 16,
    xl: 24,
  },
  typography: {
    fonts: {
      Rubik: 'Rubik',
      RubikMedium: 'Rubik-Medium',
    },
    size: {
      sm: 14,
      md: 16,
      lg: 18,
    },
  },
};

export const mockStyles = {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 24,
    backgroundColor: '#FFFFFF',
  },
  content: {
    alignItems: 'center',
    maxWidth: 300,
  },
  iconContainer: {
    width: 64,
    height: 64,
    borderRadius: 32,
    backgroundColor: '#F5F5F5',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 16,
  },
  icon: {
    fontSize: 32,
  },
  message: {
    fontFamily: 'Rubik-Medium',
    fontSize: 18,
    color: '#000000',
    textAlign: 'center',
    marginBottom: 12,
  },
  details: {
    fontFamily: 'Rubik',
    fontSize: 14,
    color: '#666666',
    textAlign: 'center',
    marginBottom: 16,
  },
  retryButton: {
    backgroundColor: '#000000',
    paddingHorizontal: 24,
    paddingVertical: 12,
    borderRadius: 12,
    marginTop: 12,
  },
  retryText: {
    fontFamily: 'Rubik-Medium',
    fontSize: 16,
    color: '#FFFFFF',
  },
};