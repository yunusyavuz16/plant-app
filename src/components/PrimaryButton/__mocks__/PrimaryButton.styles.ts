const styles = {
  container: {
    width: '100%',
    backgroundColor: '#4CAF50',
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    height: 48,
  },
  label: {
    color: '#FFFFFF',
    fontSize: 16,
    fontFamily: 'SFProText-Bold',
    letterSpacing: 0.5,
  },
};

export default styles;