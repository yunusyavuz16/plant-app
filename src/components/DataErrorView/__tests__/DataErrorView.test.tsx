import React from 'react';
import { Text, Pressable } from 'react-native';
import renderer, { act } from 'react-test-renderer';
import { mockStyles, mockTheme } from '../__mocks__/dataErrorViewStyles';

// Mock the theme constants
jest.mock('@/constants/theme', () => mockTheme);

// Mock the styles
jest.mock('../DataErrorView.style', () => ({
  styles: mockStyles,
}));

// Mock react-native-reanimated
jest.mock('react-native-reanimated', () => {
  const View = require('react-native').View;
  return {
    View,
    FadeIn: {
      duration: () => ({
        // Mock animation object
        build: () => ({
          randomId: 'mock-animation',
        }),
      }),
    },
    createAnimatedComponent: (component: any) => component,
  };
});

// Import the component after all mocks are set up
const DataErrorView = require('../DataErrorView').default;

interface DataErrorViewProps {
  message?: string;
  errorDetails?: string;
  onRetry: () => void;
}

describe('DataErrorView', () => {
  const defaultProps: DataErrorViewProps = {
    onRetry: jest.fn(),
  };

  const renderComponent = (props: DataErrorViewProps = defaultProps) => {
    let component: renderer.ReactTestRenderer;
    act(() => {
      component = renderer.create(<DataErrorView {...props} />);
    });
    return component!;
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderComponent().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with default message when no message prop is provided', () => {
    const component = renderComponent();
    const messageText = component.root.findByProps({ style: mockStyles.message });
    expect(messageText.props.children).toBe('Unable to load content');
  });

  it('renders with custom message when provided', () => {
    const customMessage = 'Custom error message';
    const component = renderComponent({ ...defaultProps, message: customMessage });
    const messageText = component.root.findByProps({ style: mockStyles.message });
    expect(messageText.props.children).toBe(customMessage);
  });

  it('renders error details when provided', () => {
    const errorDetails = 'Detailed error information';
    const component = renderComponent({ ...defaultProps, errorDetails });
    const detailsText = component.root.findByProps({ style: mockStyles.details });
    expect(detailsText.props.children).toBe(errorDetails);
  });

  it('does not render error details when not provided', () => {
    const component = renderComponent();
    const detailsTexts = component.root.findAllByProps({ style: mockStyles.details });
    expect(detailsTexts).toHaveLength(0);
  });

  it('calls onRetry when retry button is pressed', () => {
    const mockOnRetry = jest.fn();
    const component = renderComponent({ ...defaultProps, onRetry: mockOnRetry });
    const retryButton = component.root.findByType(Pressable);

    act(() => {
      retryButton.props.onPress();
    });

    expect(mockOnRetry).toHaveBeenCalledTimes(1);
  });

  it('renders with correct structure', () => {
    const component = renderComponent();

    // Check main container
    const container = component.root.findByProps({ style: mockStyles.container });
    expect(container).toBeTruthy();

    // Check content container
    const content = component.root.findByProps({ style: mockStyles.content });
    expect(content).toBeTruthy();

    // Check icon container and icon
    const iconContainer = component.root.findByProps({ style: mockStyles.iconContainer });
    expect(iconContainer).toBeTruthy();
    const icon = component.root.findByProps({ style: mockStyles.icon });
    expect(icon.props.children).toBe('⚠️');

    // Check retry button and text
    const retryButton = component.root.findByType(Pressable);
    expect(retryButton).toBeTruthy();
    const retryText = component.root.findByProps({ style: mockStyles.retryText });
    expect(retryText.props.children).toBe('Try Again');
  });

  it('applies correct styles', () => {
    const component = renderComponent();

    // Check container styles
    const container = component.root.findByProps({ style: mockStyles.container });
    expect(container.props.style).toEqual(expect.objectContaining({
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      padding: 24,
      backgroundColor: '#FFFFFF',
    }));

    // Check message styles
    const message = component.root.findByProps({ style: mockStyles.message });
    expect(message.props.style).toEqual(expect.objectContaining({
      fontFamily: 'Rubik-Medium',
      fontSize: 18,
      color: '#000000',
      textAlign: 'center',
      marginBottom: 12,
    }));

    // Check retry button styles
    const retryButton = component.root.findByProps({ style: mockStyles.retryButton });
    expect(retryButton.props.style).toEqual(expect.objectContaining({
      backgroundColor: '#000000',
      paddingHorizontal: 24,
      paddingVertical: 12,
      borderRadius: 12,
      marginTop: 12,
    }));
  });

  it('handles multiple retry attempts', () => {
    const mockOnRetry = jest.fn();
    const component = renderComponent({ ...defaultProps, onRetry: mockOnRetry });
    const retryButton = component.root.findByType(Pressable);

    // Simulate multiple presses
    act(() => {
      retryButton.props.onPress();
      retryButton.props.onPress();
      retryButton.props.onPress();
    });

    expect(mockOnRetry).toHaveBeenCalledTimes(3);
  });

  it('handles undefined onRetry prop gracefully', () => {
    // This test ensures the component doesn't crash if onRetry is undefined
    expect(() => {
      renderComponent({ onRetry: undefined as any });
    }).not.toThrow();
  });
});