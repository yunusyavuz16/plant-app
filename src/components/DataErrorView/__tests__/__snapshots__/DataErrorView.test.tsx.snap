// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataErrorView renders correctly 1`] = `
<View
  entering={
    {
      "build": [Function],
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "maxWidth": 300,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "borderRadius": 32,
          "height": 64,
          "justifyContent": "center",
          "marginBottom": 16,
          "width": 64,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 32,
          }
        }
      >
        ⚠️
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Rubik-Medium",
          "fontSize": 18,
          "marginBottom": 12,
          "textAlign": "center",
        }
      }
    >
      Unable to load content
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#000000",
          "borderRadius": 12,
          "marginTop": 12,
          "paddingHorizontal": 24,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
          }
        }
      >
        Try Again
      </Text>
    </View>
  </View>
</View>
`;
